### PR TITLE
[REVIEW] Bump Pandas support to >=1.0,<1.2.0dev0

### PR DESCRIPTION
**Pull request by brandon-b-miller**

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by brandon-b-miller** [comment]
rerun tests

**Comment by brandon-b-miller** [comment]
@galipremsagar It looks like pandas updated some of the `repr` logic around timedelta columns in 1.1, although I don't see anything about it in the patch notes. Could you take a look and make sure these timestamp column changes make sense?

**Comment by brandon-b-miller** [comment]
@kkraus14 I think the combination of repr and groupby `nunique` changes cause this PR to pass when we're at `>=1.1` but not if we're at `>=1.0, <1.1`. Do we want to just xfail those tests based on the pandas version? I'm confirming now that cuDF would be functionally the same for any pandas version in the `>=1.0, <1.2` window.

File: python/cudf/cudf/core/series.py [hunk 1]
```diff
@@ -971,15 +971,14 @@ def __repr__(self):
...
-            output = preprocess.to_pandas().__repr__()
-        elif (
+        if (
             preprocess.nullable
             and not isinstance(
                 preprocess._column, cudf.core.column.CategoricalColumn
             )
             and not is_list_dtype(preprocess.dtype)
+        ) or isinstance(
+            preprocess._column, cudf.core.column.timedelta.TimeDeltaColumn
```
Comment (galipremsagar) [inline python/cudf/cudf/core/series.py:981]:
```suggestion
```

LGTM overall, I think without this condition we should be able to run into this block if we have nulls, else we will take the else block.

Comment (brandon-b-miller) [reply python/cudf/cudf/core/series.py:981]: Yeah but I think even when there are no nulls we still need to cast to string to get the right result. Otherwise it hits `to_pandas()` first and it can change the formatting of the values. 

```
0    0 days 00:00:00.001000000
1    0 days 00:00:00.000200000
2    0 days 00:00:00.003000000
dtype: timedelta64[ns]
(Pdb) cp.asnumpy(ser._column.values)
array([1000000,  200000, 3000000], dtype='timedelta64[ns]')
(Pdb) pd.Series(cp.asnumpy(ser._column.values), dtype=ser.dtype)
0   0 days 00:00:00.001000
1   0 days 00:00:00.000200
2   0 days 00:00:00.003000
dtype: timedelta64[ns]
```

Comment (galipremsagar) [reply python/cudf/cudf/core/series.py:981]: Ah yes, you are correct. Totally forgot this difference.

**Comment by kkraus14** [comment]
> @kkraus14 I think the combination of repr and groupby `nunique` changes cause this PR to pass when we're at `>=1.1` but not if we're at `>=1.0, <1.1`. Do we want to just xfail those tests based on the pandas version? I'm confirming now that cuDF would be functionally the same for any pandas version in the `>=1.0, <1.2` window.

Can we mark those tests as xfail conditionally on the Pandas version?

File: python/cudf/cudf/tests/conftest.py [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
 def datadir():
     return pathlib.Path(__file__).parent / "data"
+
+
+@pytest.fixture(scope="session")
+def pd_version_fail():
+    if parse("1.0") <= parse(pd.__version__) < parse("1.1"):
+        return True
+    else:
+        return False
```
Comment (kkraus14) [inline python/cudf/cudf/tests/conftest.py:20]: Instead of defining a session fixture like this we should make a decorator that can be used to tag tests. Additionally we should make this a `pandas_1.0_fail` or something along those lines, and then when Pandas 1.2 comes around we could additionally have a `pandas_1.1_fail` and tag things accordingly.

**Comment by brandon-b-miller** [comment]
created a `_compat.py` to hold the constants.

**Comment by kkraus14** [comment]
Changes look good. I think we can update the conda recipes / environment files and we should be good to go.

**Comment by brandon-b-miller** [comment]
created https://github.com/rapidsai/integration/pull/113

File: conda/recipes/cudf/meta.yaml [hunk 3]
```diff
@@ -31,7 +31,7 @@ requirements:
     - cudatoolkit {{ cuda_version }}.*
   run:
     - python
-    - pandas >=1.0,<1.1.0dev0
+    - pandas >=1.0,<1.2
```
Comment (kkraus14) [inline conda/recipes/cudf/meta.yaml:34]:
```suggestion
    - pandas >=1.0,<1.2.0dev0
```

File: ci/gpu/build.sh [hunk 4]
```diff
@@ -71,8 +71,8 @@ conda install "rmm=$MINOR_VERSION.*" "cudatoolkit=$CUDA_REL" \
               "ucx-py=${MINOR_VERSION}" \
 
 # https://docs.rapids.ai/maintainers/depmgmt/
-# conda remove -f rapids-build-env rapids-notebook-env
-# conda install "your-pkg=1.0.0"
+conda remove --force rapids-build-env rapids-notebook-env
+conda install "pandas>=1.0,<1.2"
```
Comment (kkraus14) [inline ci/gpu/build.sh:75]: Revert these changes now that the upstream PR merged and we should be good to go here

Comment (brandon-b-miller) [reply ci/gpu/build.sh]: done.
